packages/headless/src/controllers/result-list/headless-result-link-analytics.ts

```diff
@@ -0,0 +1,77 @@
+import {Result} from '../../api/search/search/result';
+import {Engine} from '../../app/headless-engine';
+import {logDocumentOpen} from '../../features/result/result-analytics-actions';
+
+// 1 second is a reasonable amount of time to catch most longpress actions
+const defaultDelay = 1000;
+
+type ResultLinkAnalyticsOptions = {
+  /**
+   * The result for which to log analytics.
+   */
+  result: Result;
+  /**
+   * The amount of time to wait before selecting the result after calling `beginDelayedSelect`.
+   */
+  delay?: number;
+};
+
+type ResultLinkAnalyticsProps = {
+  /** The options for the `ResultLinkAnalytics` controller. */
+  options: ResultLinkAnalyticsOptions;
+};
+
+/**
+ * The `ResultLinkAnalytics` headless controller helps more easily send analytics when a search result is selected.
+ */
+export type ResultLinkAnalytics = ReturnType<typeof buildResultLinkAnalytics>;
```

Comment:
We'll need to extract the documentation and place it on an interface. Using `ReturnType` is not compatible with the doc-extractor tool unfortunately,

Reply:
Ah, I was waiting for your PR to be merged, but now that I see that it's merged I'll update this one.